pyaedt/edb_core/edb_data/simulation_setup.py

```diff
@@ -0,0 +1,720 @@
+        edb_setup_info = self.get_sim_setup_info
+        edb_setup_info.SimulationSettings.Enabled = value
+        self._edb_object = self._set_edb_setup_info(edb_setup_info)
+        self._update_setup()
+
+    @property
+    def name(self):
+        """Name of the setup."""
+        return self._edb_object.GetName()
+
+    @name.setter
+    def name(self, value):
+        self._pedb.layout.cell.DeleteSimulationSetup(self.name)
+        edb_setup_info = self.get_sim_setup_info
+        edb_setup_info.Name = value
+        self._name = value
+        self._edb_object = self._set_edb_setup_info(edb_setup_info)
+        self._update_setup()
+
+    @property
+    def position(self):
+        """Position in the setup list."""
+        return self.get_sim_setup_info.Position
+
+    @position.setter
+    def position(self, value):
+        edb_setup_info = self.get_sim_setup_info.SimulationSettings
+        edb_setup_info.Position = value
+        self._set_edb_setup_info(edb_setup_info)
+        self._update_setup()
+
+    @property
+    def setup_type(self):
+        """Type of the setup."""
+        return self.get_sim_setup_info.SimSetupType.ToString()
+
+    @property
+    def frequency_sweeps(self):
+        """List of frequency sweeps."""
+        temp = {}
+        for i in list(self.get_sim_setup_info.SweepDataList):
+            temp[i.Name] = EdbFrequencySweep(self, None, i.Name, i)
+        return temp
+
+    @pyaedt_function_handler
+    def _add_frequency_sweep(self, sweep_data):
+        """Add a frequency sweep.
+
+        Parameters
+        ----------
+        sweep_data: EdbFrequencySweep
+        """
+        self._sweep_list[sweep_data.name] = sweep_data
+        edb_setup_info = self.get_sim_setup_info
+
+        if self._setup_type in ["kSIwave", "kHFSS"]:
+            for _, v in self._sweep_list.items():
+                edb_setup_info.SweepDataList.Add(v._edb_object)
+
+        self._edb_object = self._set_edb_setup_info(edb_setup_info)
+        self._update_setup()
+
+    @pyaedt_function_handler
+    def delete_frequency_sweep(self, sweep_data):
+        """Delete a frequency sweep.
+
+        Parameters
+        ----------
+        sweep_data : EdbFrequencySweep
+        """
+        name = sweep_data.name
+        if name in self._sweep_list:
+            self._sweep_list.pop(name)
+
+        fsweep = []
+        for k, val in self.frequency_sweeps.items():
+            if not k == name:
+                fsweep.append(val)
+        self.get_sim_setup_info.SweepDataList.Clear()
+        for i in fsweep:
+            self.get_sim_setup_info.SweepDataList.Add(i._edb_object)
+        self._update_setup()
+        return True if name in self.frequency_sweeps else False
+
+    @pyaedt_function_handler()
+    def add_frequency_sweep(self, name=None, frequency_sweep=None):
+        """Add frequency sweep.
+
+        Parameters
+        ----------
+        name : str, optional
+            Name of the frequency sweep.
```

Comment:
```suggestion
            Name of the frequency sweep. The default is ``None``.
```
You should describe the behavior that occurs when ``None`` is used: The default is ``None``, in which case ....